addons/pkg/util/addon_util.go

```diff
@@ -218,3 +232,120 @@ func IsPackageInstallPresent(ctx context.Context,
 
 	return true, nil
 }
+
+// AddFinalizerToCRD adds finalizer to the config CRD if not present and
+// returns true if finalizer is added
+func AddFinalizerToCRD(
+	log logr.Logger,
+	addonName string,
+	configCRD client.Object) bool {
+
+	var patchAddonSecret bool
+
+	// add finalizer to addon secret
+	if !controllerutil.ContainsFinalizer(configCRD, addontypes.AddonFinalizer) {
+		log.Info("Adding finalizer to addon secret", constants.AddonNameLogKey, addonName)
+		controllerutil.AddFinalizer(configCRD, addontypes.AddonFinalizer)
+		patchAddonSecret = true
+	}
+
+	return patchAddonSecret
+}
+
+// RemoveFinalizerFromCRD removes finalizer from the config CRD if not present and
+// returns true if finalizer is removed
+func RemoveFinalizerFromCRD(
+	log logr.Logger,
+	addonName string,
+	configCRD client.Object) bool {
+
+	var patchAddonSecret bool
+
+	// add finalizer to addon secret
+	if !controllerutil.ContainsFinalizer(configCRD, addontypes.AddonFinalizer) {
+		log.Info("Removing finalizer to addon secret", constants.AddonNameLogKey, addonName)
+		controllerutil.RemoveFinalizer(configCRD, addontypes.AddonFinalizer)
+		patchAddonSecret = true
+	}
+
+	return patchAddonSecret
+}
+
+// get paths for external CRDs by introspecting versions of the go dependencies
+func GetExternalCRDPaths(externalDeps map[string][]string) ([]string, error) {
+	var crdPaths []string
+	gopath, err := exec.Command("go", "env", "GOPATH").Output()
+	if err != nil {
+		return crdPaths, err
+	}
+	for dep, crdDirs := range externalDeps {
+		depPath, err := exec.Command("go", "list", "-m", "-f", "{{ .Path }}@{{ .Version }}", dep).Output()
+		if err != nil {
+			return crdPaths, err
+		}
+		for _, crdDir := range crdDirs {
+			crdPaths = append(crdPaths, filepath.Join(strings.TrimSuffix(string(gopath), "\n"),
+				"pkg", "mod", strings.TrimSuffix(string(depPath), "\n"), crdDir))
+		}
+	}
+
+	logf.Log.Info("external CRD paths", "crdPaths", crdPaths)
+	return crdPaths, nil
+}
+
+// This function returns the Service CIDR blocks for both IPv4 and IPv6 family
+// Parse Service CIDRBlocks obtained from the cluster and return the respective blocks for IPv4 and IPv6 family
+// Return "" if we don't find the CIDRBlocks for an IP family
+func GetServiceCIDRs(cluster *clusterapiv1beta1.Cluster) (string, string, error) {
```

Comment:
@vijaykatam, @shyaamsn  , PTAL. I have introduced this new function for deriving serviceCIDR and serviceCIDRv6 fields for Antrea